src/main/java/frc/robot/systems/DriveFSMSystem.java

```diff
@@ -323,12 +324,41 @@ public void drive(double xSpeed, double ySpeed, double rot,
 		rearRight.setDesiredState(swerveModuleStates[(2 + 1)]);
 	}
 
+	/**
+	 * Drives robot to specific position and orientation.
+	 * @param input Global TeleopInput if robot in teleop mode or null if
+	 *        the robot is in autonomous mode.
+	 * @param xFinal desired x position in meters
+	 * @param yFinal desired y position in meters
+	 * @param rotFinal desired rotation in degrees
+	 *
+	 */
+	public void driveToPos(TeleopInput input, double xFinal, double yFinal, double rotFinal) {
+		double xDist = xFinal - getPose().getX();
+		double yDist = yFinal - getPose().getY();
+		double rotAmount = rotFinal - (getHeading() % THREE_SIXTY_DEGREES);
+
+		if (Math.abs(xDist) < AutoConstants.METERS_MARGIN_OF_ERROR
+			&& Math.abs(yDist) < AutoConstants.METERS_MARGIN_OF_ERROR
+			&& Math.abs(rotAmount) < AutoConstants.DEGREES_MARGIN_OF_ERROR) {
+			drive(0, 0, 0, false, false);
+		} else {
+			double xSpeed = xDist / AutoConstants.DRIVE_TO_TAG_TRANSLATIONAL_CONSTANT;
```

Comment:
I think for swerve you want to control for the overall speed, not the speed of the individual axes. In other words, the magnitude of the velocity vector. Otherwise you'd be sqrt(2) faster if the direction to the target happens to be on a diagonal.

Reply:
It shouldn't matter because our x and y components of this speed vector are proportional to how far we need to move to get to our target coordinates, right? Because the drive() function in our swerve method accepts the xSpeed, ySpeed, and angularSpeed as parameters, we can't simply use the magnitude.